core/src/main/scala/org/http4s/jdkhttpclient/JdkWSClient.scala

```diff
@@ -129,6 +129,15 @@ object JdkWSClient {
                       })
                     } yield ()
                   }
+
+              shouldAbort <- F.delay {
+                !webSocket.isOutputClosed || !webSocket.isInputClosed
+              }
+
+              _ <- if(shouldAbort) F.delay {
+                webSocket.abort()
+              } else F.unit
```

Comment:
I wonder if we should place this inside a `.guarantee { ... }`? And also can't we put the whole thing inside of a `delay { ... }` ?

Reply:
Turns out you were right initially. Part of the issue is not waiting for the close frame from the server, the abort like this can lead to broken pipe errors on server side. I'll need to do a bit more thinking. In our code I'm playing around with waiting on the `closedDef` for some amount of time in case the input is still open before i do the emergency-abort.